src/NLPModels.jl

```diff
@@ -419,20 +420,83 @@ function jac_op!(nlp :: AbstractNLPModel, x :: AbstractVector, rows :: AbstractV
   return jac_op!(nlp, rows, cols, vals, Jv, Jtv)
 end
 
+"""
+    Hv = jth_hprod(nlp, x, v, j)
+
+Evaluate the product of the j-th Hessian of the constraints with the vector `v`.
+"""
 function jth_hprod(nlp::AbstractNLPModel, x::AbstractVector, v::AbstractVector, j::Integer)
   @lencheck nlp.meta.nvar x v
-  hv = Vector{eltype(x)}(undef, nlp.meta.nvar)
-  return jth_hprod!(nlp, x, v, j, hv)
+  @assert 0 ≤ j ≤ nlp.meta.ncon
+  Hv = Vector{eltype(x)}(undef, nlp.meta.nvar)
+  return jth_hprod!(nlp, x, v, j, Hv)
 end
 
+"""
+    Hv = jth_hprod!(nlp, x, v, j, Hv; obj_weight=1.0)
+
+Evaluate the product of the j-th Hessian of the constraints with the vector `v`
+in place.
+"""
 function jth_hprod! end
 
+"""
+    vals = jth_hess_coord(nlp, x, j; obj_weight=1.0)
+
+Evaluate the j-th Hessian at `x` scaled by `obj_weight` in sparse coordinate 
+format.
+Only the lower triangle is returned.
+"""
+function jth_hess_coord(nlp::AbstractNLPModel, x::AbstractVector, j::Integer; obj_weight = one(eltype(x)))
+ @lencheck nlp.meta.nvar x
+ @assert 0 ≤ j ≤ nlp.meta.ncon
+ vals = Vector{eltype(x)}(undef, nlp.meta.nnzh)
+ return jth_hess_coord!(nlp, x, j, vals; obj_weight=obj_weight)
+end
+
+"""
+    vals = jth_hess_coord!(nlp, x, j, vals; obj_weight=1.0)
```

Comment:
And here.

Reply:
What should be the size of `vals` here? Is there an assumption that the same overall `nnzh` is used for all constraint Hessians?